### DISCUSSION TITLE
IRHVAC not supported on ESP32?

### DISCUSSION THREAD

**Discussion by augea** [arendst/Tasmota#18256] [Q&A]
I use IRHVAC for several years on ESP8266 hardware. Now I decided to build a sensor board with all features I need based a newer ESP32. I integrated also an IR power LED to be independent of weak transmitters. 
I tried to compiled my own firmware with all drivers, but were not successful with sending IRHVAC commands.
I also tested tasmota32-ir but also no IRHVAC commands are available.
CMD: IRHVAC {"Vendor":"MITSUBISHI_AC"}
RSL: RESULT = {"Command":"Unknown"}
I am able to receive IR command but these are also shown as unknown protocol.
{"IrReceived":{"Protocol":"UNKNOWN","Bits":90,"Hash":"0xED762824","Repeat":0}}

Does ESP32 branch not support IRHVAC commands? 
Is this planned for the future or is this a bug in Tasmota?

Used software:
Program Version  12.4.0(ir)

**Comment by barbudor** [comment]
> I tried to compiled my own firmware with all drivers

Please elaborate

> Program Version 12.4.0

Please try latest development
There have been some fixes on IR for ESP32 recently but unsure if these are yet included in 12.4.0

**Reply by augea** [reply]
@barbudor Just tested with "Development binaries for Tasmota firmware 12.4.0.4 on ESP32"  (tasmota32-ir.bin).
With this development firmware was able to receive and decoded signals and was able to send commands successfully.

Thank you for your very quick and extremely helpful response. I will use the development software for further testing my board.
